services/sendemail.js

```diff
@@ -1,61 +1,68 @@
-const nodemailer = require("nodemailer");
-const hbs = require("nodemailer-express-handlebars");
-const env = require('../configuration/env');
-const fs = require("fs");
-
-require('dotenv').config()
+const User = require('../models/user');
+const crypto = require('crypto');
 
 exports.sendEmail = async (email, templateName) => {
-    
-    let transporter = nodemailer.createTransport({
-      host: env.SMTP_HOST,
-      port: env.SMTP_PORT,
-      secure: env.SMTP_SECURE,
-      auth: {
-        user: env.SMTP_USER,
-        pass: env.SMTP_PASS,
+  let transporter = nodemailer.createTransport({
+    host: env.SMTP_HOST,
+    port: env.SMTP_PORT,
+    secure: env.SMTP_SECURE,
+    auth: {
+      user: env.SMTP_USER,
+      pass: env.SMTP_PASS,
+    },
+  });
+
+  transporter.use(
+    'compile',
+    hbs({
+      viewEngine: {
+        extName: '.hbs',
+        partialsDir: './email-templates/',
+        layoutsDir: './email-templates/',
+        defaultLayout: '',
       },
+      viewPath: './email-templates/',
+      extName: '.hbs',
     })
-  
-    transporter.use(
-      'compile',
-      hbs({
-          viewEngine: {
-              extName: '.hbs',
-              partialsDir: './email-templates/',
-              layoutsDir: './email-templates/',
-              defaultLayout: '',
-          },
-          viewPath: './email-templates/',
-          extName: '.hbs',
-      })
   );
 
-  const templatePath = `./email-templates/${templateName}.hbs`
+  const templatePath = `./email-templates/${templateName}.hbs`;
   const template = fs.readFileSync(templatePath, 'utf-8');
   const emailSubject = getSubject(template);
 
-    const msg = {
-        from: '"Ant Colony - project pharaoh" project.pharaoh@hotmail.com', // sender address
-        to: `${email}`, // list of receivers
-        subject: emailSubject, // Subject line
-        template: templateName //template
-    }
-    // send mail with defined transport object
-    const info = await transporter.sendMail(msg);
+  // generate reset token
+  const resetToken = crypto.randomBytes(20).toString('hex');
+  const resetTokenExpirationDate = new Date(Date.now() + 24 * 60 * 60 * 1000); // token is valid for 24 hours
+
+  // save resetToken and resetTokenExpirationDate to database
+  const user = await User.findOneAndUpdate({ email: email }, { resetToken: resetToken, resetTokenExpirationDate: resetTokenExpirationDate });
+
+  if (!user) {
+    throw new Error(`User with email ${email} not found`);
+  }
 
-    console.log("Message sent: %s", info.messageId);
+  const msg = {
+    from: '"Ant Colony - project pharaoh" project.pharaoh@hotmail.com', // sender address
+    to: `${email}`, // list of receivers
+    subject: emailSubject, // Subject line
+    template: templateName, //template
+    context: {
+      resetLink: `https://example.com/reset-password?token=${resetToken}&email=${email}`,
```

Comment:
I think Meliha do this to be reusable, and you all you need to do is to reuse it. to call this function in service and pass all arguments. you don't need to correct something